media/rtmp2segment.go

```diff
@@ -186,6 +186,9 @@ func processSegments(segmentHandler SegmentHandler, outFilePattern string, compl
 	pipeNum := 0
 	createNamedPipe(fmt.Sprintf(outFilePattern, pipeNum))
 
+	// shared buffer; set to empty and segment reader will initialize
+	buf := []byte{}
```

Comment:
**nit:** Could consider using a `bytes.NewBuffer(nil)` as well. With some existing impl like [`ReadFrom`](https://cs.opensource.google/go/go/+/go1.23.4:src/bytes/buffer.go;l=206) we can avoid manually handling buffer sizes/increases etc. 

That would involve a refactor tho so def not a must. Just leaving the idea mostly for the future.

Reply:
Yep we need to switch to a buffered reader soon to decouple reads from writes, since the right now this can lead to back-pressure if RTMP segment reads out-pace trickle writes.

With that being said, this specific change might end up being redundant after all, so I think I'll just discard the commit entirely (theres a merge conflict with it right now anyway)

Reply:
Yeah the `bytes.Buffer` wouldn't solve back pressure as it would return an EOF after each segment. I'm always susprised how there's no buffered pipe built-in in Go. Shouldn't be hard to make one tho